### DISCUSSION TITLE
Consider using Taffy for Layout

### DISCUSSION THREAD

**Discussion by RichiCoder1** [vadimdemedes/ink#555] [Ideas]
[Taffy](https://github.com/DioxusLabs/taffy) is a pretty awesome lib in development with the intent of building something like Yoga to support more layout styles like Grid, Table, and other layout styles seamlessly.

Would be great to see Ink use it both for Grid support and slightly more web-complaint Flexbox implementation, as well as other layout algorithms as they're implemented.

This likely would not be a small change as 1) I don't think Taffy yet has an NAPI package and 2) it would change the behavior of consumers code (possibly).

**Comment by sindresorhus** [comment]
Taffy looks awesome. I would much rather depend on some Rust code than C++ (Yoga). It's not clear how stable it is though. Yoga has years of maturity.

---

> This likely would not be a small change as 1) I don't think Taffy yet has an NAPI package and 2) it would change the behavior of consumers code (possibly).

We would not want to use NAPI anyway. Rust code is usually easy to convert to WASM, which is what we want.

**Reply by RichiCoder1** [reply]
> It's not clear how stable it is though. Yoga has years of maturity.

That's absolutely fair for sure. 

> We would not want to use NAPI anyway. Rust code is usually easy to convert to WASM, which is what we want.

That makes sense and Nico, maintainer, also mentioned they'd like to prioritize C/WASM first so hopefully that happens soon! 

https://github.com/DioxusLabs/taffy/issues/392#issuecomment-1464985055
https://github.com/DioxusLabs/taffy/issues/241
https://github.com/DioxusLabs/taffy/issues/345

Apparently someone undertook Deno bindings via wasm (see https://github.com/load1n9/gelatin) but I'm not sure how adaptable it'd be.
